lib/server.js

```diff
@@ -264,7 +264,34 @@ function Client(stream, socket) {
   }
   // end service/auth-related ==================================================
 
+  var unrepliedGlobalRequests = [];
+
+  function sendReplies() {
+    var reply;
+    while (unrepliedGlobalRequests[0] && unrepliedGlobalRequests[0].type) {
```

Comment:
I think the first part of this conditional should instead be `unrepliedGlobalRequests.length > 0` to avoid an out-of-bounds access which could cause a performance hit.